src/core/thread/mle.cpp

```diff
@@ -2056,6 +2061,13 @@ otError Mle::SendDataRequest(const Ip6::Address &aDestination,
     SuccessOrExit(error = AppendActiveTimestamp(*message));
     SuccessOrExit(error = AppendPendingTimestamp(*message));
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
```

Comment:
Suggest removing this `#if` since we are in general adding these paramaters to the `SendDataRequest()`.

Another part in th code may use it without realizing that it is tied to the `OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE`.

Another option is to also add the parameters to the method definition conditionally.